lib/src/io/node.dart

```diff
@@ -80,6 +90,22 @@ _readFile(String path, [String encoding]) {
   }
 }
 
+Future<String> readStdin() async {
+  var completer = new Completer<String>();
+  var contents = new StringBuffer();
+  _stdin.on('data', allowInterop((buf) {
+    contents.write(UTF8.decode(buf));
```

Comment:
This runs the risk of splittting a character across chunk boundaries. You can use `Converter.startChunkedConversion()` to track state across chunks, probably with `StringConversionSink.withCallback()` to get the end result.

Also this should probably be `SystemEncoding`.